src/common/com/intellij/plugins/haxe/ide/surroundWith/HaxeDoWhileSurrounder.java

```diff
@@ -37,7 +38,8 @@
   protected PsiElement doSurroundElements(PsiElement[] elements, PsiElement parent) {
     final HaxeDoWhileStatement whileStatement =
       (HaxeDoWhileStatement)HaxeElementGenerator.createStatementFromText(elements[0].getProject(), "do {\n} while(a);");
-    addStatements(whileStatement.getBlockStatement(), elements);
+    List<HaxeBlockStatement> blockStatementList = whileStatement.getBlockStatementList();
+    addStatements(blockStatementList.get(0), elements);
```

Comment:
Yes, I filed a bug on the Jetbrains/grammar-kit repo and they haven't addressed it.